### DISCUSSION TITLE
Can cosserat rods be used for simulating robot cable?

### DISCUSSION THREAD

**Discussion by DebbieDebbieli** [GazzolaLab/PyElastica#272] [General]
Hi,
Recently, I am working on simulating the cable dressed up on robot arms. The cable moves following with the robot and may be twisted.
At the beginning I try the spring-mass system to model the cable. The result is unsatisfied. Are cosserat rods adept on this?
![image](https://github.com/GazzolaLab/PyElastica/assets/16498562/a673a2c7-a5d3-4589-a55c-cc9acec570bd)
(https://www.youtube.com/watch?v=Jl8ygt8qRtU)

**Comment by armantekinalp** [comment]
Hi @DebbieDebbieli 

Yes you can use `PyElastica` to simulate cables. PyElastica is a Cosserat Rod solver and it captures all 6 modes of deformation bending (2 direction), shear (2 direction), twist and stretch.

**Reply by DMCardenas99** [reply]
Hi @armantekinalp 
In these cases where we have a **start** and **end** connector with their specific **directions**, what approach is recommended to make the simulation?
I've been working on a similar project but I'm still struggling.  I'm defining an approximated position in a CAD software and obtaining the "spline points". Since the proposed hose position is not real, I'm trying to use pyelastica to get the realistic position of the hose.
Is there any example of a similar situation?

Thank you in advance for your help.
